service/src/main.rs

```diff
@@ -881,16 +915,36 @@ fn enclave_account<E: EnclaveBase>(enclave_api: &E) -> AccountId32 {
 }
 
 /// Checks if we are the first validateer to register on the parentchain.
-fn we_are_primary_validateer(
+fn we_are_primary_worker(
```

Comment:
this logic has now changed to something more meaningful. Instead of testing if we are the first enclave to have ever registered on this chain ( which would allow only a single sidechain to ever exist) we now check shard_status and see if some other enclave than self has ever touched that shard